doc/setup.md

```diff
@@ -165,3 +165,29 @@ make load_demo_data
 ```
 
 This also adds `testuser@cesium-ml.org` as an administrator.
+
+### Deploying secure HTTP / SSL certificate
+
+When running a public server, you will likely want to deploy an SSL certificate (i.e., serve `https://your.url` instead of `http://your.url`). Certificates can be obtained for free from services such as Let's Encrypt (https://letsencrypt.org/). For Let's Encrypt at least, the process requires verification of the service without SSL first (i.e. simply running `make run`).
+
+Briefly, one can install certbot:
+    pip install certbot-nginx
+
+and retrieve a certificate:
+    sudo certbot certonly --standalone --preferred-challenges http -d http://your.url
+or similar if using https
+    sudo certbot certonly --standalone --preferred-challenges https -d https://your.url
+
+To renew the certificate, one does:
+    sudo certbot renew
+
+After receiving an SSL certificate, you'll need to modify the nginx configuration in `baselayer/services/nginx/nginx.conf.template` to use it. For example, you would add the following for Let's Encrypt:
+
+    listen [::]:443 ssl ipv6only=on;
+    listen 443 ssl;
```

Comment:
Did you try this? I think the config is a bit convoluted, and may require a more sophisticated patch (or, at least, a patch to the right place in the config file).

Reply:
Yeah this is what I am running for sedmv2 currently (copied and pasted). But it doesn't mean that it's a good choice!

Reply:
We currently have:

```
  server {
    listen {{ ports.app }};
```

So it's not quite clear to me where the above goes. Did you replace the `listen` statements?

Reply:
No I left those in and just copied and pasted the below into the template. The other listens are still in there. We still want the app ports I presume?

Reply:
Can you paste the entire config file here?